src/modules/login_user/app/login_user_controller.ts

```diff
@@ -0,0 +1,43 @@
+
+import { MissingParameters, WrongTypeParameters } from '../../../shared/helpers/errors/controller_errors'
+import { EntityError } from '../../../shared/helpers/errors/domain_errors'
+import { IRequest } from '../../../shared/helpers/external_interfaces/external_interface'
+import { BadRequest, InternalServerError, OK, NotFound } from '../../../shared/helpers/external_interfaces/http_codes'
+import { LoginUserUsecase } from './login_user_usecase'
+import { LoginUserViewmodel } from './login_user_viewmodel'
+import { NoItemsFound } from '../../../shared/helpers/errors/usecase_errors'
+
+export class LoginUserController {
+  constructor(private usecase: LoginUserUsecase) {}
+
+  async handle(request: IRequest) {
+    try {
+      const { email, password } = request.data as { email: string, password: string }
+
+      if (!email || !password) {
```

Comment:
I think it would be better if it was two separate validations for email and password
And also add an WrongTypeParameter validations for each